cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/worker/sidecar/ArtifactLocalizerHttpHandlerInternal.java

```diff
@@ -16,44 +16,47 @@
 
 package io.cdap.cdap.internal.app.worker.sidecar;
 
-import com.google.gson.Gson;
-import com.google.gson.GsonBuilder;
 import com.google.inject.Inject;
 import com.google.inject.Singleton;
-import io.cdap.cdap.common.conf.CConfiguration;
 import io.cdap.cdap.common.conf.Constants;
-import io.cdap.cdap.proto.BasicThrowable;
-import io.cdap.cdap.proto.codec.BasicThrowableCodec;
+import io.cdap.cdap.proto.id.ArtifactId;
 import io.cdap.http.AbstractHttpHandler;
 import io.cdap.http.HttpHandler;
 import io.cdap.http.HttpResponder;
 import io.netty.handler.codec.http.HttpRequest;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import io.netty.handler.codec.http.HttpResponseStatus;
 
+import java.io.File;
+import javax.ws.rs.DefaultValue;
 import javax.ws.rs.GET;
 import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.QueryParam;
 
 /**
- * Internal {@link HttpHandler} for File Localizer.
+ * Internal {@link HttpHandler} for Artifact Localizer.
  */
 @Singleton
 @Path(Constants.Gateway.INTERNAL_API_VERSION_3 + "/worker")
 public class ArtifactLocalizerHttpHandlerInternal extends AbstractHttpHandler {
-  private static final Logger LOG = LoggerFactory.getLogger(ArtifactLocalizerHttpHandlerInternal.class);
-  private static final Gson GSON = new GsonBuilder()
-    .registerTypeAdapter(BasicThrowable.class, new BasicThrowableCodec()).create();
+  private final ArtifactLocalizer artifactLocalizer;
 
   @Inject
-  public ArtifactLocalizerHttpHandlerInternal(CConfiguration cConf) {
-    super();
-
+  public ArtifactLocalizerHttpHandlerInternal(ArtifactLocalizer artifactLocalizer) {
```

Comment:
Remove `public` for injected constructor.

Reply:
This needs to be public so that the `RemoteConfiguratorTest` can see it. I added `@VisibleForTesting` to make that more clear. 

This is one of the issues that was causing unit tests to fail after I added the `ArtifactLocalizerClient` to the `RemoteArtifactrRepositoryReader`. Since the `RemoteConfiguratorTest` doesnt use any Guice bindings we need to manually instantiate the ArtifactLocalizerHttpHandler

Reply:
Actually I don't see this class being binded in guice anywhere. What it has `@Inject` annotation??

Reply:
Removed the `@Inject`, it wasnt needed